### DISCUSSION TITLE
Solinteg support

### DISCUSSION THREAD

**Discussion by ziat007** [wills106/homeassistant-solax-modbus#1144] [General]
I am hopefully getting a new solar installation soon.

So far the best offer (the most reliable solar company) is recommending the Solinteg MHT-10K-25 inverter. (allegedly due to its very good efficiency, low power loss).

After reading around here and GitHub, I have not found very much support.

Solinteg claims they natively support Modbus standard, with their own wifi-lan/Modbus adapter for use with third-party apps.

Can anybody confirm or deny this?

MODBUS/TCP adapter 
https://www.solinteg.com/wp-content/uploads/2023/07/DuoCOM-Wi-FiLAN-2-in-1-Module-Introduction20230728.pdf

Solinteg 3rd party "documentation"
https://www.solinteg.com/wp-content/uploads/2024/05/Solinteg-Third-Party-Device-Combination.pdf

Possible detailed information:
https://eshop.helion.cz/user/related_files/solinteg_modbus_protocol_mht-25-50.pdf

**Comment by wills106** [comment]
The following Inverters are currently supported:

https://github.com/wills106/homeassistant-solax-modbus/blob/a4453600cce0d79c654de25de25dbc4342368717/custom_components/solax_modbus/plugin_solinteg.py#L99-L134

@gec75 Added in support for Solinteg, personally I don't really know anything about them.

**Reply by ziat007** [reply]
Thank you very much!

**Comment by gec75** [comment]
Hi,
Yes, I also have a Solinteg inverter (MHT-20K-40).
The support was pretty good, they also listened to some fixes/optimizations for the firmware.
I implemented the plugin based on their Modbus documentation. I did NOT implement all the settings/features which one can do via App/Web, but the most important.
They also tested a little the plugin and they were quite satisfied.
Otherwise I am not affiliated in any way with them.

Best regards, Gabriel

**Reply by ziat007** [reply]
What connection are you using/do you recommend?
Preferably if I do need an external adaptor I would like to use the one directly from Solinteg.

I was told by the IT department of my solar installation company that HA can connect directly to the invertor, although I have not found any mention of this.

Installation should be done in around 3 months.

**Reply by gec75** [reply]
I am using their LAN adapter, I didn't try whether it also works over Wifi.
Yes, HA connects directly to the inverter (you have to find the IP address).

**Reply by tomihatvan** [reply]
> I am using their LAN adapter, I didn't try whether it also works over Wifi. Yes, HA connects directly to the inverter (you have to find the IP address).

Dear Gabriel, could you write/show me your configuration? I tried it several times, several ways without a luck.. :(

**Reply by gec75** [reply]
I don't know exactly what configuration you need...
You can use this tool to test the connection: https://github.com/ClassicDIY/ModbusTool
![image](https://github.com/user-attachments/assets/8fd8adba-a819-4a32-8656-e612f2366be4)
You should set the port (502), ip address, connect.
Then try to read below; at address 10000 should return the serial number...

**Reply by tomihatvan** [reply]
that's what I wanted to find :) The port number is 502 and the modbus address is 255

**Reply by gec75** [reply]
Sorry, I didn't think about it :)
502 is standard MobusTCP port, modbus address is indeed needed.

**Reply by tomihatvan** [reply]
I am a noob :D so it was a really helpful information. The point is that I was able to integrate the inverter into my home assitant system.

**Reply by gec75** [reply]
So it also works over wifi, yes?
I'm glad you solved it.

**Reply by tomihatvan** [reply]
No, I had to buy a duocom module, which is able to communicate via wifi and ethernet

**Comment by tomihatvan** [comment]
Hi guys!

I also tried to setup a 5K25 via Wifi ( probably I found the ip address, but something is wrong. Can you show me your config? I didn't find it in the description. Thank you in advance!!

**Comment by ziat007** [comment]
I am a little confused. Is there any reason this shouldnt work over wifi?
I can not connect.  The IP of the official dongle shows a Wifi Assist/OTA screen.

when I attempt to connect with smodified settings of:


TCP / Ethernet
modbus address: 255
ivertor type: solinteg
IP: 192.168.0.XXX
TCP port: 502

I get a setup failed.

Log:
https://gist.github.com/ziat007/36d553a23030ab7213e36b9228d7c2a3

**Reply by LAMARKO71** [reply]
Pomocí Wifi to nelze ovládat. Na Solintegu mi sdělili že potřebuji adaptér který používá LAN.
![6672_solinteg-integ-r-series-lan-modul](https://github.com/user-attachments/assets/e854000c-4012-463d-9b17-8c6ae697a490)

**Reply by ziat007** [reply]
I am still a little lost. I have a LAN module now. The inverter seems to be connected to the app. LAN module light is blue.

There seem to be 2 new DHCP clients, both espressif. (and the previous inverter itself? Shanghai High-Flying Electronics Technology Co., Ltd)
When visiting the two new addresses via HTTP they both display a WIFI assistant page.

When trying to connect to either of these addresses with modbus via HA I get the same error as previously.

TCP / Ethernet
modbus address: 255
invertor type: solinteg
IP: 192.168.0.XXX
TCP port: 502

**Comment by gec75** [comment]
First just try to see if it responds:
telnet IP 502

Otherwise I also used ModbusTool for testing
https://github.com/ClassicDIY/ModbusTool

**Reply by ziat007** [reply]
It does not respond.
According to the app it is one of the two suspected IP's, but no response on `telnet 192.168.0.69 502`
`Connecting To 192.168.0.69...Could not open connection to the host, on port 502: Connect failed`

HTTP displays WIFI assistant website.


Modbus Master, TCP, 192.168.0.69, Port 502, Slave ID 255
Machine actively refused it.

**Comment by gec75** [comment]
I think it might be the Wifi module, mine (LAN) does not respond on http.
And I read somewhere that only the LAN module responds on ModbusTCP port.

**Reply by ziat007** [reply]
Is there any way I can identify the Lan module IP?
What is is that I have 3 DHCP clients that are suspected solinteg.

**Reply by gec75** [reply]
Just try each one with telnet :)

**Reply by ziat007** [reply]
I have multiple times.
I am a little lost on where or what troubleshoot or where to look for documentation.

Internal network traffic should be wide open. (opnsense)

**Reply by tomihatvan** [reply]
Or set your unit to fix ip. It was the solution in my case.

**Comment by ziat007** [comment]
The issue was Solinteg inverter was connected via Wifi (I have the LAN dongle)
Changing its IP it assigned the LAN dongle for communication.

Its visible in the integtech app, under device -> device name-> Bassic info -> scroll down to communication mode, should say LAN-RS485

**Reply by miszterx** [reply]
Hi,
I have exactly the same situation. Same inverter, same dongle (wifi+LAN). On the plastic cover there is reset button, but no button on the card itself. I have a SW2 switch on the board.
It was setup with WIFI and on the solinteg app i can see: "Communication mode: WIFI-/ "

Even if I connect to LAN cable, I have different IP but even on the IP of the cable connection loads the Wifi Assistant / OTA page.
Is there a way how you can switch? Or how you could bypass this situation? I am also lookiong for make available the 502 tcp port for modbus2tcp for further integration.

**Reply by ziat007** [reply]
In your DHCP leases there should be two solinteg clients with the name Espressif Inc.

I reassigned them both to a static IP and that fixed the issue for me.

I hope it will work for you.

**Reply by gec75** [reply]
Maybe (temporarily) disable Wifi? Or change the password?
I am an "early adopter" with separate adapters, only LAN is connected.

**Reply by miszterx** [reply]
Great! Thanks for the Help! For me the ultimate solution was to restart the inverter itself. Then it came up with LAN only mode. Now I have 502 port as well:

telnet 10.4.4.179 502                                                                                                                           ─╯
Trying 10.4.4.179...
Connected to 10.4.4.179.
Escape character is '^]'.

Even a dummy modbus client said successfully connected. Whatever it means.
However if I try to add the HACS's SolaX Inverter Modbus and add the IP, port 502, type solinteg and modbus address 1. I get only error:

Failed to set up:
Check the logs

When I check the local logs:

```
2025-07-07 21:10:57.076 ERROR (MainThread) [pymodbus.logging] No response received after 6 retries, continue with next request
2025-07-07 21:10:57.076 ERROR (MainThread) [custom_components.solax_modbus] Error: device: 1 address: 10008 -> Modbus Error: [Input/Output] No response received after 6 retries, 
continue with next request
2025-07-07 21:10:57.076 WARNING (MainThread) [custom_components.solax_modbus.plugin_solinteg] solinteganother: attempt to read model failed at 0x2718
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/plugin_solinteg.py", line 93, in _read_model
    if not data.isError():
           ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'isError'
2025-07-07 21:10:57.077 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry solinteganother for solax_modbus
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 749, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/__init__.py", line 208, in async_setup_entry
    await hub.async_init()
  File "/config/custom_components/solax_modbus/__init__.py", line 342, in async_init
    self._invertertype = await self.plugin.async_determineInverterType(self, self.config)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/plugin_solinteg.py", line 1294, in async_determineInverterType
    bh, bl = model // 256, model % 256
             ~~~~~~^^~~~~
TypeError: unsupported operand type(s) for //: 'NoneType' and 'int'
```

Have you see similar? or which HA integration and input parameters you used to successfully setup?

**Reply by gec75** [reply]
Modbus address is 255

**Reply by miszterx** [reply]
Thanks for the hint. Unfortunately it is still error:

```
2025-07-07 21:39:32.690 ERROR (MainThread) [custom_components.solax_modbus] Error: device: 255 address: 10000 -> Modbus Error: [Connection] Not connected[AsyncModbusTcpClient 10.4.4.179:502]
2025-07-07 21:39:32.690 WARNING (MainThread) [custom_components.solax_modbus.plugin_solinteg] solinteg: attempt to read serialnumber failed at 0x2710
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/plugin_solinteg.py", line 71, in _read_serialnr
    if not data.isError():
           ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'isError'
2025-07-07 21:39:32.690 WARNING (MainThread) [custom_components.solax_modbus.plugin_solinteg] solinteg: reading serial number from address 0x2710 failed; other address may succeed
2025-07-07 21:39:32.690 ERROR (MainThread) [custom_components.solax_modbus.plugin_solinteg] solinteg: cannot find serial number, even not for other Inverter
2025-07-07 21:39:32.691 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('10.4.4.179', 502)
2025-07-07 21:39:32.792 WARNING (MainThread) [custom_components.solax_modbus] Unable to connect to Inverter at 10.4.4.179:502
2025-07-07 21:39:32.792 ERROR (MainThread) [custom_components.solax_modbus] Error: device: 255 address: 10008 -> Modbus Error: [Connection] Not connected[AsyncModbusTcpClient 10.4.4.179:502]
2025-07-07 21:39:32.792 WARNING (MainThread) [custom_components.solax_modbus.plugin_solinteg] solinteg: attempt to read model failed at 0x2718
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/plugin_solinteg.py", line 93, in _read_model
    if not data.isError():
           ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'isError'
2025-07-07 21:39:32.793 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry solinteg for solax_modbus
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 749, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/__init__.py", line 208, in async_setup_entry
    await hub.async_init()
  File "/config/custom_components/solax_modbus/__init__.py", line 342, in async_init
    self._invertertype = await self.plugin.async_determineInverterType(self, self.config)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/plugin_solinteg.py", line 1294, in async_determineInverterType
    bh, bl = model // 256, model % 256
             ~~~~~~^^~~~~
```
Is it a fix number BTW or something what I can check on the inverter?

**Reply by ziat007** [reply]
> Maybe (temporarily) disable Wifi? Or change the password? I am an "early adopter" with separate adapters, only LAN is connected.

I have the Lan only adapter. At least that is what is said on the box. Somehow the ivnertor still was able to connect via WiFi.

I even have 2 wireless leases connected from the inverter.
And a direct WiFi hotspot available too.


All showing the Splinteg WiFi/firmware setup page.

**Reply by miszterx** [reply]
Hey guys! not sure why or what happened. but i have restarted the inverter again. Then it came up and with modbus address 255 it added. Thanks for your support!

**Reply by ziat007** [reply]
> Hey guys! not sure why or what happened. but i have restarted the inverter again. Then it came up and with modbus address 255 it added. Thanks for your support!

It took me 4 days of restarting everything until changing the DHCP helped me.

The connection behaviours seems to be odd. Assume the app info now says connected via Lan. (that seemed to be the teltale sign)

**Comment by ziat007** [comment]
How is it possible to control Battery discharging into the grid and charging from the grid? 10K-25

Via the App and modbus it is done via setting the inverter to Economic mode and setting it in the submenu of economic mode. I can set it to economic mode in the integration, but there is no submenu.

**Reply by ziat007** [reply]
@gec75 as you implemented it I assumed you might know.

**Reply by gec75** [reply]
Yes, didn't implement any time-related (submenus) settings...

**Reply by ziat007** [reply]
> Yes, didn't implement any time-related (submenus) settings...

So is there a way to force charge the battery and to force grid export without them?
There is a setting "Feed-In" inverter working mode, but it does nothing.

I am assuming for force charging setting the work mode to UPS would work?

**Reply by ziat007** [reply]
A workaround is:
General mode as standard.
UPS mode as forced charging.
Economy mode - in the app set up all times to be discharging.

This way you can set all 3 behaviour via HA.

**Comment by gec75** [comment]
Yes, that sounds good.
I don't aim to build a complete replacement for the app...

**Reply by ziat007** [reply]
I understand. What I was primarily after is being able to control battery charge rate, force charge, force discharge.

**Comment by dvbhannes** [comment]
Try EVCC. Set up the inverter with Modbus or SunSpec, and you can automatically charge the battery at specific times when electricity is cheap.

**Comment by Jorn-H** [comment]
Just out of curiosity, as I am trying to setup my solinteg inverter (MHT-20K-40) with home assistant:

Did you get this integration working, as I am having some troubles communicating with the inverter via the integration.

**Reply by miszterx** [reply]
Hi! Yep I could manage it. I have a MHT-5K-25, but I think integration wise it would be the same. I made a short blog post on it, which can guide you. It is unfortunately in hungarian. https://blog.xorp.hu/solinteg-mht-5k-25-modbus-home-assistant-integration/ Drop it to google translate to translate the content. I tried to add screenshots as well. Thre are some key points. e.g.: proper adapter type, use only LAN, be sure you have switched to lan, use the proper attributes, then it will work.  If you have specific question just ping me and I hope I can help.

**Reply by Jorn-H** [reply]
Thanks a lot, so the modbus port is actually 255, while the inverter says 247...

I have not gotten it to work yet, but is it possible to force charge/discharge with home assistant via this integration?

**Reply by gec75** [reply]
It should work...
Known details:
- TCP port number 502 (standard)
- modbus address is 255
- works only with LAN, not with WIFI !?

What exactly do you mean force charge/discharge? There are several ways to do it... Min SOC, Working mode (UPS), etc...

**Comment by dvbhannes** [comment]
What settings have you made on the inverter and the integration? Is the inverter connected to the network via LAN?

**Comment by gec75** [comment]
Your question is too generic, do you want to know something specific?
I implemented most functionality in the integration, and I have lots of settings.
Yes, the inverter is connected via LAN. I think it doesn't work with Wifi.

**Comment by Jorn-H** [comment]
As of now, I am just starting with this. Did not really work on it too much yet. 

Currently having troubles with connecting to it, have found the IP (which solinteg doesn't show, instead it says: �������������). Trying to connect to it over TCP, but can not control or read anything (error: Actie button/press kon niet worden uitgevoerd. Modbus Error: [Connection] Not connected[AsyncModbusTcpClient 192.168.0.50:502]) 

Do In have to manually set something in the inverter to enable modbus over TCP?

Besides, which version of the integration is known to work?

**Comment by dvbhannes** [comment]
Besides, which version of the integration is known to work?

the current version works

05.09.2025 18:43:38 Jorn-H ***@***.***>:

**Comment by Jorn-H** [comment]
Hello again,

Below I have posted pictures of my current settings in the integration:

First, as shown here, the local ip of the inverter is currently 192.168.0.50
<img width="951" height="36" alt="afbeelding" src="https://github.com/user-attachments/assets/1296feb1-faed-40a3-afb0-71b4539a5680" />

Than, the first input menu of the solax inverter modbus integration:
<img width="493" height="1241" alt="afbeelding" src="https://github.com/user-attachments/assets/fcb7ec48-9584-448e-be2f-7866a323995c" />

Followed by the last menu:
<img width="405" height="544" alt="afbeelding" src="https://github.com/user-attachments/assets/f669f1aa-8d54-4b75-a251-125c88e823d1" />

When adding the inverter now, it will not give any errors. It does not read out any specs of the inverter however. Using the action buttons like turning of, starting or restarting the inverter via the integration return the previously posted error code:
(error: Actie button/press kon niet worden uitgevoerd. Modbus Error: [Connection] Not connected[AsyncModbusTcpClient 192.168.0.50:502])

My bad for the fact that parts of the pictures and error code are dutch, but it should be clear what they say without translation.

Do I have something configured wrong?

**Comment by infradom** [comment]
@Jorn-H : Ising a broadcast address 255 is usually not a good idea; it may be used for initial discovery, but I believe it is better to check the inverter's real addrass and use its normal address (like 1 up to 254). This is normally configurable somewhere in the inverter's config menu (I dont know Solinteg at all)

**Comment by gec75** [comment]
I think 255 is the Modbus address, should be ok.
As I wrote earlier (https://github.com/wills106/homeassistant-solax-modbus/discussions/1144#discussioncomment-13505301), try to use Modbus Tool for testing.

**Reply by Jorn-H** [reply]
It looks like I can not reach the inverter this way either. Is this know to work with the standard lan adapter (see picture)
![s](https://github.com/user-attachments/assets/44366347-f6a7-46c6-81d7-8e3f639d3aeb)


Or only with the duo com
![g](https://github.com/user-attachments/assets/02163fab-f516-4a8d-8bde-d4bb3f5dac11)

**Comment by gec75** [comment]
I have a LAN adapter, and it's known it doesn't work with wifi.
It looks like a networking problem... does "ping" work?

**Reply by Jorn-H** [reply]
I can indeed ping the IP, but am currently at work (over VPN connection). Will try again tonight.

Just for clarity, I will describe the installation of the networking. My modem is connected directly to a switch, which holds the ethernet connection to the ethernet adapter of the inverter. The inverter is in the same vlan as my mini pc, running proxmox (which has a VM with HAOS).

Am I right to think that there is not really anything wrong here? (I can add other simpler smart devices without problems).

**Comment by Jorn-H** [comment]
After finally receiving an installer account for the solinteg cloud server, I was able to update the firmware (which was V01.00). This solved the problem and I can now connect to the inverter flawlesly.

Thanks for the help :)

**Reply by JoachimAndersson** [reply]
Where did you find update firmware ?

**Reply by Jorn-H** [reply]
This is a 'feature' that is part of an installer account for the cloud environment. You should ask solinteg directly for s installer account.

**Reply by gec75** [reply]
I also have an installer account...
If you can't solve it, you can add my account as "installer" and I can trigger the update

**Reply by Jorn-H** [reply]
Same for me, but would advice to get the installer account yourself. The installer basically has full control of the system.

(Not like we would misuse your inverter, but I wanted to mention it at least)

**Reply by JoachimAndersson** [reply]
Thanks. Just to ask theie support for one ? Do you need to enable modbus or it is on by default ?

**Reply by Jorn-H** [reply]
You can ask the technical support mail of Solinteg, you do not need to enable modbus. (Modbus is enabled by default and is not used by the web interface)

**Reply by gec75** [reply]
I was involved from the beginning with Solinteg for modbus TCP support.
At the beginning it was kind of beta firmware, but now it's always included.
But it requires wired connection, it doesn't work over wifi.

**Reply by rapsus** [reply]
hi guys, in my case, it is quite complicated to set up a LAN connection from the duoCOM to the router. Is there any device I can buy to avoid having to run cables?

**Reply by gec75** [reply]
You need some kind of wireless bridge... like TP-Link AC750?

**Comment by Jorn-H** [comment]
@gec75 I have a question regarding adding extra parameters to the integration.

I have managed to do this, except for all parameters regarding the economic mode period settings. When setting a value on these, they automatically revert to their previous settings. I have don the following:

Added under SELECT_TYPES:

    SolintegModbusSelectEntityDescription(
        name="Period 1 dis_charge",
        key="per1_dis_charge",
        register=53007,
        option_dict={
            0x000: "None",
            0x001: "Charge",
            0x002: "Discharge",
        },
        entity_category=EntityCategory.CONFIG,
        allowedtypes=HYBRID,
        icon="mdi:dip-switch",
    ),

Added under SENSOR_TYPES:

    SolintegModbusSensorEntityDescription(
        key="per1_dis_charge_rd",
        register=53007,
        scale={
            0x000: "None",
            0x001: "Charge",
            0x002: "Discharge",
        },
        internal=True,
        # allowedtypes = HYBRID,
    ),

This would be one of the multiple values, but they all behave the same.

Out of your head, can you see anythin that could be wrong here? (I am just starting to get into this stuff, have a light background in some coding from school. Please don't bash me too hard or go light with the programming 'slang') ':)

**Comment by gec75** [comment]
You have to use the same key.
But other than that, periods might need first the working mode change, etc...

**Comment by gurkburk76** [comment]
So i've managed to install the plugin and it's talking to my 20k-40 solinteg, step 1 complete. I feel the app leaves some stuff to be desired so i might take a stab at creating some "logic" of my own, but i'de appreciate some form of basis to work off from, @gec75 (or anyone else) do you have any ha/node-red flow or something that you'de like to share so i can get my feet wet? :)

**Reply by ziat007** [reply]
I am using predbat to control my solar system. It is very configurable and supports a huge amount of features.

**Reply by Jorn-H** [reply]
@ziat007

Would you care to share how you have set this up?

**Reply by ziat007** [reply]
@Jorn-H 
here is my apps.yaml for predbat.

Solinteg is not officially supported, but this integration is.

There might still be a few hiccups in my predbat config for solinteg. I wanted to get it merged as supported, but I just haven't had the time.
https://gist.github.com/ziat007/c463e2c72b6f538204658187a4db7e2e

**Reply by gurkburk76** [reply]
I've been reading up a bit on predbat and it seems to want to use givtcp and with that solinteg is not supported, do i understand it correctly when i think that you pretty much ignore givtcp and instead use this (solax) integration? that would make sense.

**Reply by ziat007** [reply]
@gurkburk76 
Predbat has multiple control methods for invertors.

if you read the configuration I sent then you will see what I am using with solinteg.

as scheduling isn’t supported with solax integration I use direct real time control.

Now that u have polished a few bugs in the config it keeps writes to a minimum.

**Comment by rapsus** [comment]
Do you think that the limitation of only being able to connect via LAN is an insurmountable limitation, or could an update by Solinteg allow us to connect to Modbus via Wi-Fi?

**Reply by gec75** [reply]
it is definitely artificial, but if they want to change... I have no idea.

**Reply by ziat007** [reply]
I have mine on a tp link nano

**Reply by dvbhannes** [reply]
Plug the TPLink directly into the inverter's USB port???

**Reply by ziat007** [reply]
To link as a wireless bridge (it connects to you WiFi)
-> Ethernet from tplink into duocom or lan module.

**Reply by dvbhannes** [reply]
So don't plug the TPlink Nano into the inverter's USB port...

**Reply by rapsus** [reply]
It would be fantastic if the manufacturer offered the option of connecting via Wi-Fi. It is absurd to need another device for this.

**Reply by ziat007** [reply]
I also recommend setting predbat to monitor mode at first, to see everything is working correctly.

**Reply by rapsus** [reply]
> I also recommend setting predbat to monitor mode at first, to see everything is working correctly.

I don't know how to do this :(

**Reply by ziat007** [reply]
Predbat has verry good documentation, browse though it or use ai to assist you to browse through it.

https://springfall2008.github.io/batpred/
